### DISCUSSION TITLE
YAML static asset import support

### DISCUSSION THREAD

**Discussion by JulianCataldo** [withastro/roadmap#200] [Proposal]
Hello there,

I might miss something but for now, the only way to statically import YAML data, as you would with JSON, is to convert it in Markdown first, with `---` fences, then only loading and using frontmatter data.

Amaright ?

**Comment by delucis** [comment]
Hi @JulianCataldo — that would work, yes!

Otherwise you could load and parse using a package like [`yaml`](https://www.npmjs.com/package/yaml):

```astro
---
import { parse } from 'yaml';
import { readFileSync } from 'node:fs';

const rawText = readFileSync('../data.yml', 'utf-8');
const data = parse(rawText);
---
<pre>{JSON.stringify(data, null, 2)}</pre>
```

You could also load multiple `.yml` files using [`Astro.glob()`](https://docs.astro.build/en/reference/api-reference/#astroglob).

**Reply by JulianCataldo** [reply]
Yeah, that's the natural work-around I would choose if I couldn't use the Yaml > MD trick.
I prefer to not rely too much on direct file I/O methods, keeps thing more future proof and inline with other static + native import methods.
Thanks a lot !

**Reply by delucis** [reply]
Right — `Astro.glob` would be the better static import way. (I said you could use it for multiple files, but `Astro.glob('../data.yml')` works too, returning an array with just that file.)

**Reply by JulianCataldo** [reply]
Ok, now I understand the second part of your first answer :)
Didn't know `Astro.glob` could load YML directly. Singleton array is no big deal. Thanks !
Also, I thought `Astro.glob` was a dynamic import (with `import()` under the hood), but still, a native ES6 way (most important).
